pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -957,10 +951,12 @@ private void closeConsumerTasks() {
     }
 
     private void failPendingReceive() {
-        if (pinnedExecutor != null && !pinnedExecutor.isShutdown()) {
-            failPendingReceives(this.pendingReceives);
-            failPendingBatchReceives(this.pendingBatchReceives);
-        }
+        internalPinnedExecutor.execute(() -> {
+            if (pinnedExecutor != null && !pinnedExecutor.isShutdown()) {
```

Comment:
Testing for IaShutdown is probably not enough, because in the meantime someone may shutdown it.
Probably we can add try/catch RejectedExecutionException

Reply:
But this queue is unbounded. If the thread pool is closed, can't throw an exception directly?

Reply:
I think we can add a UnCatchExceptionHandler to log errors